fendermint/vm/interpreter/src/fvm/state/exec.rs

```diff
@@ -235,6 +235,23 @@ where
         self.execute_message(msg, ApplyKind::Implicit)
     }
 
+    pub fn execute_read_only(&mut self, msg: Message) -> ExecResult {
+        if let Err(e) = msg.check() {
+            return Ok(check_error(e));
+        }
+
+        let raw_length = fvm_ipld_encoding::to_vec(&msg).map(|bz| bz.len())?;
+        let ret = self.executor.execute_message_with_revert(
+            msg,
+            ApplyKind::Implicit,
+            raw_length,
+            true,
+        )?;
+        let addrs = self.emitter_delegated_addresses(&ret)?;
+
+        Ok((ret, addrs))
+    }
+
     /// Execute message implicitly but ensures the execution is successful and returns only the ApplyRet.
     pub fn execute_implicit_ok(&mut self, msg: Message) -> ExecResult {
```

Comment:
```suggestion
     /// Execute the message implicitly and ensure a successful application.
    ///
    /// This function wraps `execute_implicit` but inspects the result.
    /// If any failure information is present in the result, it returns an error.
    /// (Renamed from `execute_implicit_ok` for clarity.)
    pub fn execute_implicit_or_fail(&mut self, msg: Message) -> ExecResult {
```

Reply:
I try to contain the PR, but the name change can be a separate PR instead?